src/middlewared/middlewared/plugins/network.py

```diff
@@ -148,6 +148,22 @@ def network_config_extend(self, data):
         )
     )
     async def do_update(self, data):
+        """
+        Update Network Configuration Service configuration.
+
+        `ipv4gateway` if set is used instead of the default gateway provided by DHCP.
+
+        `nameserver1` is primary DNS server.
+
+        `nameserver2` is secondary DNS server.
+
+        `nameserver3` is tertiary DNS server.
+
+        `httpproxy` attribute is provided if a proxy is to be configured for Middlewared operations.
```

Comment:
I haven't seen context, but I suspect it can be fixed by eliminating the reference to middlewared and the strange use of "configured" to mean used. Also, be definite about whether it has to be provided or not:
The 'httpproxy' attribute must be provided if a proxy is used.

Reply:
@wblock i have updated it with your suggestion but the issue with middlewared still stands. So if proxy is provided, all middlewared operations use that proxy, if a user was to use CLI and expect the proxy to be respected, that would not happen. I think we should outline that here. Please let me know what you think. Thank you

Reply:
s/middlewared/network/ and we should be good. Lets not mention middlewared, please.